### PR TITLE
HWKALERTS-182 Update swagger doc

### PR DESCRIPTION
**Pull request by lucasponce**
HKWALERTS-184 Wrap deleted number response in a json
A necessary update on the swagger doc.
Now data types are covered and API operations reviewed.
Unified some style as readers of swaggers are REST API users (non Java users).
Fixing minor non-json types for consistency.
Several tricks on the asciidoc templates as swagger has some limitations for Java types or free text.

### CODE REVIEW

File: hawkular-alerts-api/src/main/java/org/hawkular/alerts/api/model/event/Event.java [hunk 1]
```diff
@@ -57,51 +67,105 @@
...
+            "to evaluate events with triplet [tenantId, source, dataId] as unique identifier.",
+            position = 4,
+            allowableValues = "Timestamp in milliseconds.")
     @JsonInclude
     String dataSource;
 
     // Optional dataId of Event, used by EventCondition to evaluate events
+    @ApiModelProperty(value = "Data identifier used for Events condition evaluation. All Events participating on " +
+            "conditions must have a valid dataId. DataId in events context can be read as an events source name. + \n" +
+            "Events generated from <<Trigger>> will use the triggerId as dataId.",
```
Comment (jshaughn) [inline hawkular-alerts-api/src/main/java/org/hawkular/alerts/api/model/event/Event.java]: Suggested:
'Data identifier used for Events condition evaluation. All Events participating on conditions must have a valid dataId. DataId in events context can be read as an events source name. Events generated from <<Trigger>> will use the triggerId as dataId.'
-->
'Data identifier used for Event condition evaluation.  Events must supply a valid dataId to be considered for EventCondition evaluation.  DataIds in an events context should incorporate the source of the event (for uniqueness). Events generated from a <<Trigger>> will have dataId set to the triggerId, therefore allowing chaining with other EventConditions.'

If you don't take the suggestion you must minimally: 'events' -> 'event's where used in the possessive sense.

File: hawkular-alerts-api/src/main/java/org/hawkular/alerts/api/model/trigger/Trigger.java [hunk 2]
```diff
@@ -31,99 +31,189 @@
...
 
     /** Resolve all unresolved alerts when auto-resolve condition-set is satisfied */
+    @ApiModelProperty(value = "Resolve all unresolved alerts when auto-resolve condition-set is satisfied.",
+            position = 15,
+            example = "false")
     @JsonInclude
     private boolean autoResolveAlerts;
 
+    @ApiModelProperty(value = "The policy used for deciding whether the trigger auto-resolved condition-set is " +
+            "satisfied. Either ALL conditions must evaluate to true, or whether ANY one is enough.",
```
Comment (jshaughn) [inline hawkular-alerts-api/src/main/java/org/hawkular/alerts/api/model/trigger/Trigger.java]: Suggestion:
'Either ALL conditions must evaluate to true, or whether ANY one is enough.'
->
'ALL conditions must evaluate to true or ANY one condition must evaluate to true.'

File: hawkular-alerts-api/src/main/java/org/hawkular/alerts/api/model/trigger/Trigger.java [hunk 3]
```diff
@@ -31,99 +31,189 @@
...
     private String memberOf;
 
+    @ApiModelProperty(value = "A enabled trigger is loaded into the engine for data evaluation.",
+            position = 19,
+            example = "false")
     @JsonInclude
     private boolean enabled;
 
+    @ApiModelProperty(value = "The policy used for deciding whether the trigger condition-set is satisfied. " +
+            "Either ALL conditions must evaluate to true, or whether ANY one is enough.",
```
Comment (jshaughn) [inline hawkular-alerts-api/src/main/java/org/hawkular/alerts/api/model/trigger/Trigger.java]: Suggestion:
'Either ALL conditions must evaluate to true, or whether ANY one is enough.'
->
'ALL conditions must evaluate to true or ANY one condition must evaluate to true.'

File: hawkular-alerts-api/src/main/java/org/hawkular/alerts/api/model/trigger/Trigger.java [hunk 4]
```diff
@@ -31,99 +31,189 @@
...
+            position = 20,
+            example = "ALL")
     @JsonInclude
     private Match firingMatch;
 
+    @ApiModelProperty(value = "Extended mechanism to match trigger's condition against Data with [source, dataId] " +
```
Comment (jshaughn) [inline hawkular-alerts-api/src/main/java/org/hawkular/alerts/api/model/trigger/Trigger.java]: 'trigger's condition' -> 'trigger conditions'

File: hawkular-alerts-api/src/main/java/org/hawkular/alerts/api/model/trigger/TriggerAction.java [hunk 5]
```diff
@@ -40,26 +43,55 @@
  * @author Jay Shaughnessy
  * @author Lucas Ponce
  */
+@ApiModel(description = "Link an <<ActionDefinition>> with a <<Trigger>>. + \n" +
+        " + \n" +
+        "It can add optional constraints that determine when an action will be executed: + \n" +
+        " + \n" +
+        "- A set of Alert.Status (represented by its string value). + \n" +
+        "The action will be executed if the Alert which is linked is on one of the states defined. + \n" +
```
Comment (jshaughn) [inline hawkular-alerts-api/src/main/java/org/hawkular/alerts/api/model/trigger/TriggerAction.java]: Suggestion:
'The action will be executed if the Alert which is linked is on one of the states defined. '
-->
'The action will be executed if the linked Alert is set to one of the defined states. '

File: hawkular-alerts-rest/hawkular-alerts-rest-api/src/main/java/org/hawkular/alerts/rest/AlertsHandler.java [hunk 6]
```diff
@@ -386,52 +402,58 @@ public Response deleteAlert(
...
             final Long startTime,
-            @ApiParam(required = false, value = "Filter out alerts created after this time, millisecond since epoch.")
+            @ApiParam(required = false, value = "Filter out alerts created after this time.",
+                allowableValues = "Timestamp in millisecond since epoch.")
             @QueryParam("endTime")
             final Long endTime,
-            @ApiParam(required = false, value = "Filter out alerts for unspecified alertIds, " +
-                    "comma separated list of alert IDs.")
+            @ApiParam(required = false, value = "Filter out alerts for unspecified alertIds.",
+                allowableValues = "Comma separated list of trigger IDs.")
```
Comment (jshaughn) [inline hawkular-alerts-rest/hawkular-alerts-rest-api/src/main/java/org/hawkular/alerts/rest/AlertsHandler.java]: 'trigger' -> 'alert'

File: hawkular-alerts-api/src/main/java/org/hawkular/alerts/api/model/condition/CompareCondition.java [hunk 7]
```diff
@@ -44,12 +51,22 @@
     @JsonInclude(Include.NON_NULL)
     private String dataId;
 
+    @ApiModelProperty(value = "Compare operator [LT (<), GT (>), LTE (<=), GTE (>=)].",
+            position = 0,
+            required = true)
     @JsonInclude(Include.NON_NULL)
     private Operator operator;
 
+    @ApiModelProperty(value = "Data identifier of the metric used for comparation.",
```
Comment (jshaughn) [inline hawkular-alerts-api/src/main/java/org/hawkular/alerts/api/model/condition/CompareCondition.java]: 'comparation' -> 'comparison'

File: hawkular-alerts-api/src/main/java/org/hawkular/alerts/api/model/condition/ExternalCondition.java [hunk 8]
```diff
@@ -32,6 +34,17 @@
  * @author Jay Shaughnessy
  * @author Lucas Ponce
  */
+@ApiModel(description = "An ExternalCondition is used for condition evaluations performed outside of the " +
+        "Alerts engine. + \n" +
```
Comment (jshaughn) [inline hawkular-alerts-api/src/main/java/org/hawkular/alerts/api/model/condition/ExternalCondition.java]: optional: 'Alerts' -> 'Alerting'

File: hawkular-alerts-api/src/main/java/org/hawkular/alerts/api/model/condition/ExternalCondition.java [hunk 9]
```diff
@@ -32,6 +34,17 @@
...
+        " + \n" +
+        "The external engine will send StringData providing the data for which the external evaluation " +
+        " + \n" +
+        "has already evaluated to true. + \n" +
+        " + \n" +
+        "The Alerts engine assumes a true evaluation for the data being sent in from the external engine. " +
```
Comment (jshaughn) [inline hawkular-alerts-api/src/main/java/org/hawkular/alerts/api/model/condition/ExternalCondition.java]: optional: 'Alerts' -> 'Alerting'

**Comment by lucasponce** [comment]
Thanks for the review.

**Pull request merged**